### PR TITLE
Correct fetch_balance

### PR DESCRIPTION
**Pull request by magisterRab**

### CODE REVIEW

File: src/prop_tree.cpp [hunk 1]
```diff
@@ -140,7 +140,8 @@ ptree prop_list(const client::history::list& rows,
 
     for (const auto& row: rows)
     {
-        received = ceiling_add(received, row.value);
+        if (row.output.hash() != null_hash)
+            received = ceiling_add(received, row.value);
 
         if (row.spend.hash() != null_hash)
             spent = ceiling_add(spent, row.value);
```
Comment (evoskuil) [inline src/prop_tree.cpp]: Please explain the error that you are attempting to correct. This looks incorrect, as every row assumes an output that may or may not be spent. If there is no output the result should be a sentinel indicating missing necessary data. The missing data can be the result of a server code error or data corruption, or may be the result of a partially-indexed store. But if there is no output for a given input, suppressing that information is undesired behavior.

**Comment by magisterRab** [comment]
Or you can set a fixed number on error. Now the result changes from launch to launch, which is misleading

**Comment by evoskuil** [comment]
> Or you can set a fixed number on error.

That is the effect of ```ceiling_add(received, row.value);``` when ```row.output.hash() == null_hash``` as ```row.value``` must be max. As you can see in your outputs, this sentinel is ```18446744073709551615``` as a base 10 value.

**Comment by magisterRab** [comment]
I meant this code

```
for (const auto& row: rows)
     {
        if (row.output.hash() == null_hash) {
            received = 18446744073709551615;
            spent = 18446744073709551615;
            break;
       }
```

Now the output in fetch-balance turns out anything, except for 18446744073709551615

**Comment by evoskuil** [comment]
The output sum is inconsequential if the spend indicates an uncorrelated value.

**Comment by evoskuil** [comment]
@magisterRab Maybe I'm misinterpreting your comments. Is there a scenario where an uncorrelated value does not result in a `value` sentinel as documented?

**Comment by evoskuil** [comment]
There is a well documented case of the server not properly recording address history due to write concurrency and the same is possible due to lack of multi-signature address extraction (previously documented), and non-standard scripts will not record transfer against an address. In these cases bx detects lack of correlation if there is a missing output and sets a sentinel `value` so that the caller is made aware of the limitation. This does not result from code in bx but from libbitcoin-client, which correlates history received from the server. The intended behavior is that the `spent` value is max int in the case of a miscorrelation, which as far as I can tell is operating properly. We should re-investigate if this is not the case, but otherwise this change is partly redundant with libbitcoin-client code and otherwise inconsequential (as described above).

**Pull request closed without merging**